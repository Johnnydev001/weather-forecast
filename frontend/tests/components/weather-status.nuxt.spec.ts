import { mountSuspended } from '@nuxt/test-utils/runtime';
import { shallowMount } from '@vue/test-utils';
import {it, describe, expect, vi} from 'vitest'
import { a } from 'vitest/dist/chunks/suite.B2jumIFP.js';
import WeatherStatusComponent from '~/components/weather-status/weather-status.vue';


describe('Weather status tests', () => {


    it('should match the snapshot when not redirected', async () => {

        const props = 'paris';

        const weatherStatus = await mountSuspended(WeatherStatusComponent, {
            data() {
                return {
                    address: {
                        country: 'FR',
                        city: 'Paris',
                        name: 'Paris'
                    },
                    lat: 10,
                    lon: 10
                }
            },
            props: {
                locationToFind: props
            },
           
        })

console.log(weatherStatus.html())
        expect(weatherStatus.html()).toMatchSnapshot()


    })

})