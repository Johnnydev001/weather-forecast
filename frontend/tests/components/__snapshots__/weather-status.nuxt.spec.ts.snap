// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Weather status tests > should match the snapshot when not redirected 1`] = `
"<section data-v-c02f7a4d="" class="container" style="background-image: url(&quot;/assets/imgs/background-sunny.jpg&quot;);">
  <section data-v-c02f7a4d="" class="sub-container">
    <aside data-v-c02f7a4d="" role="alert" class="error-msg">Cannot find any path matching /api/weather?weatherType=current.</aside>
    <div data-v-c02f7a4d="" class="location-container">
      <div data-v-c02f7a4d=""><span data-v-c02f7a4d="" class="location-city"></span><span data-v-c02f7a4d="" class="location-country"></span></div><span data-v-c02f7a4d="" class="date">Today, 0, November</span>
    </div><img data-v-c02f7a4d="" alt="" data-nuxt-img="" srcset="/_ipx/_/assets/imgs/sun.svg 1x, /_ipx/_/assets/imgs/sun.svg 2x" class="weather-illustration" title="" src="/_ipx/_/assets/imgs/sun.svg">
    <section data-v-c02f7a4d="" class="summary-container">
      <h3 data-v-c02f7a4d="" title="Temperature. Units - metric: Celsius" class="temperature" style="color: rgb(100, 177, 255);">0ยบ </h3>
      <div data-v-c02f7a4d="" class="feels-like-temperature-container"><span data-v-c02f7a4d="">Feels like:</span>
        <h4 data-v-c02f7a4d="" title="Feels like temperature. Units - metric: Celsius" class="feels-like-temperature" style="color: rgb(100, 177, 255);">0ยบ</h4>
      </div>
    </section>
    <article data-v-c02f7a4d="" class="weather-container">
      <div data-v-c02f7a4d="" class="condition-container">
        <div data-v-c02f7a4d="" class="condition" title="The current wind speed in km/h"><svg data-v-c02f7a4d="" xmlns="http://www.w3.org/2000/svg" width="36" height="36" viewBox="0 0 24 24" fill="rgb(100, 177, 255)" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="ai ai-Air">
            <path d="M3 8h7a3 3 0 1 0-3-3"></path>
            <path d="M4 16h11a3 3 0 1 1-3 3"></path>
            <path d="M2 12h17a3 3 0 1 0-3-3"></path>
          </svg>
          <div data-v-c02f7a4d="" class="value"><span data-v-c02f7a4d="" class="title">Wind speed</span><span data-v-c02f7a4d="">0 KM/H</span></div>
        </div>
        <div data-v-c02f7a4d="" class="condition" title="The current humidity of the air in %"><svg data-v-c02f7a4d="" viewBox="0 0 24 24" fill="rgb(50, 150, 255)" stroke="rgb(100, 177, 255)" xmlns="http://www.w3.org/2000/svg">
            <g id="SVGRepo_bgCarrier" stroke-width="1rem"></g>
            <g id="SVGRepo_tracerCarrier" stroke-linecap="round" stroke-linejoin="round"></g>
            <g id="SVGRepo_iconCarrier">
              <path d="M12 21.5C16.1012 21.5 19.5 18.4372 19.5 14.5714C19.5 12.1555 18.2672 9.71249 16.8732 7.70906C15.4698 5.69214 13.8515 4.04821 12.9778 3.21778C12.4263 2.69364 11.5737 2.69364 11.0222 3.21779C10.1485 4.04821 8.53016 5.69214 7.1268 7.70906C5.73282 9.71249 4.5 12.1555 4.5 14.5714C4.5 18.4372 7.8988 21.5 12 21.5Z" stroke="rgb(100, 177, 255)"></path>
              <path d="M12 18C11.4747 18 10.9546 17.8965 10.4693 17.6955C9.98396 17.4945 9.54301 17.1999 9.17157 16.8284C8.80014 16.457 8.5055 16.016 8.30448 15.5307C8.10346 15.0454 8 14.5253 8 14" stroke="" stroke-linecap="round"></path>
            </g>
          </svg>
          <div data-v-c02f7a4d="" class="value"><span data-v-c02f7a4d="" class="title">Humidity</span><span data-v-c02f7a4d="">0 %</span></div>
        </div>
        <div data-v-c02f7a4d="" class="condition" title="Atmospheric pressure on the sea level, hPa"><svg data-v-c02f7a4d="" viewBox="0 0 24 24" fill="#ffffff" stroke="" xmlns="http://www.w3.org/2000/svg">
            <g id="SVGRepo_bgCarrier" stroke-width="0"></g>
            <g id="SVGRepo_tracerCarrier" stroke-linecap="round" stroke-linejoin="round"></g>
            <g id="SVGRepo_iconCarrier">
              <path d="M13 13L16 8M8 8H8.01M12 6H12.01M18 12H18.01M6 12H6.01M14 15C14 16.1046 13.1046 17 12 17C10.8954 17 10 16.1046 10 15C10 13.8954 10.8954 13 12 13C13.1046 13 14 13.8954 14 15ZM21 12C21 16.9706 16.9706 21 12 21C7.02944 21 3 16.9706 3 12C3 7.02944 7.02944 3 12 3C16.9706 3 21 7.02944 21 12Z" stroke="orangered" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"></path>
            </g>
          </svg>
          <div data-v-c02f7a4d="" class="value"><span data-v-c02f7a4d="" class="title">Pressure</span><span data-v-c02f7a4d="">0 hPa</span></div>
        </div>
        <div data-v-c02f7a4d="" class="condition" title="The current UV index"><svg data-v-c02f7a4d="" xmlns="http://www.w3.org/2000/svg" width="36" height="36" viewBox="0 0 24 24" fill="orange" stroke="red" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="ai ai-Sun">
            <path d="M12 3V2m0 20v-1m9-9h1M2 12h1m15.5-6.5L20 4M4 20l1.5-1.5M4 4l1.5 1.5m13 13L20 20"></path>
            <circle cx="12" cy="12" r="4" stroke="orange"></circle>
          </svg>
          <div data-v-c02f7a4d="" class="value"><span data-v-c02f7a4d="" class="title">UV index</span><span data-v-c02f7a4d="">0</span></div>
        </div>
      </div>
      <!--v-if-->
    </article>
  </section>
</section>"
`;
